components/org.wso2.carbon.identity.api.server.script.library/org.wso2.carbon.identity.api.server.script.library.v1/src/main/java/org/wso2/carbon/identity/api/server/script/library/v1/core/ServerScriptLibrariesService.java

```diff
@@ -283,10 +285,18 @@ private ScriptLibraryResponse createScriptLibraryResponse(FunctionLibrary script
         ScriptLibraryResponse scriptLibraryResponse = new ScriptLibraryResponse();
         scriptLibraryResponse.setName(scriptLibrary.getFunctionLibraryName());
         scriptLibraryResponse.setDescription(scriptLibrary.getDescription());
-        scriptLibraryResponse.setContentRef(ContextLoader.buildURIForBody(String.format(
-                V1_API_PATH_COMPONENT + SCRIPT_LIBRARY_PATH_COMPONENT + "/%s" + SCRIPT_LIBRARY_CONTENT_PATH,
-                scriptLibrary.getFunctionLibraryName())).toString());
-        return scriptLibraryResponse;
+        try {
+            String displayName =
+                    URLEncoder.encode(scriptLibrary.getFunctionLibraryName(), StandardCharsets.UTF_8.name());
+            scriptLibraryResponse.setContentRef(ContextLoader.buildURIForBody(String.format(
+                    V1_API_PATH_COMPONENT + SCRIPT_LIBRARY_PATH_COMPONENT + "/%s" + SCRIPT_LIBRARY_CONTENT_PATH,
+                    displayName)).toString().replace("+", "%20"));
+            return scriptLibraryResponse;
+        } catch (UnsupportedEncodingException e) {
+            FunctionLibraryManagementException error = new FunctionLibraryManagementException(
+                    Constants.ErrorMessage.ERROR_CODE_ERROR_ENCODING_URL.getMessage(), e);
+            throw handleScriptLibraryError(error, Constants.ErrorMessage.ERROR_CODE_ERROR_ENCODING_URL);
```

Comment:
ErrorResponse errorResponse =
                    new ErrorResponse.Builder().withCode(Constants.ErrorMessage.ERROR_CODE_ERROR_ENCODING_URL.getCode()).withMessage(Constants.ErrorMessage.ERROR_CODE_ERROR_ENCODING_URL.getMessage())
                .withDescription(Constants.ErrorMessage.ERROR_CODE_ERROR_ENCODING_URL.getDescription()).build();
            throw new APIError(Response.Status.INTERNAL_SERVER_ERROR, errorResponse);

Reply:
This fix is not required since `handleScriptLibraryError` method can access even the access modifier is private.